src/hipscat/pixel_tree/pixel_tree_builder.py

```diff
@@ -132,17 +132,126 @@ def create_node_and_parent_if_not_exist(self, pixel: HealpixInputTypes, node_typ
 
         self.create_node(pixel, node_type, parent)
 
-    def create_node(self, pixel: HealpixInputTypes, node_type: PixelNodeType, parent: PixelNode):
+    def create_node(
+            self,
+            pixel: HealpixInputTypes,
+            node_type: PixelNodeType,
+            parent: PixelNode = None,
+            replace_existing_node=False,
+    ):
         """Create a node and add to `self.pixels` in the tree
 
         Args:
             pixel: HEALPix pixel to get. Either of type `HealpixPixel`
                 or a tuple of (order, pixel)
             node_type: Node type of the node to create
-            parent: Parent node of the node to create
+            parent: (Default: None) Parent node of the node to create. If None, will get parent from
+                existing node in tree at correct position based on pixel
+            replace_existing_node: (Default: False) If True and node at `pixel` already exists,
+                the existing node will be removed and replaced with the new node.
         """
         pixel = get_healpix_pixel(pixel)
+        if parent is None:
+            if pixel.order == 0:
+                parent = self.root_pixel
+            else:
+                parent_pixel = pixel.convert_to_lower_order(delta_order=1)
+                if parent_pixel not in self:
+                    raise ValueError(
+                        f"Cannot create node at {str(pixel)}, "
+                        f"no parent node exists at {str(parent_pixel)}"
+                    )
+                parent = self[parent_pixel]
+        if parent.node_type == PixelNodeType.LEAF:
```

Comment:
Yeah, I actually initially had that logic in here but moved it out because I thought they were two different operations. I think it's probably neater this way, but it might end up being useful to have an option for it in `create_node`